src/transformers/cache_utils.py

```diff
@@ -239,8 +250,8 @@ def update(
         """
         # Optional kwargs for `SinkCache` -- needed on models using RoPE. `partial_rotation_size` is used on models
         # with partially rotated position embeddings, like Phi or Persimmon.
-        sin = cache_kwargs.get("sin")
-        cos = cache_kwargs.get("cos")
+        sin = cache_kwargs.get("sin")[: self.window_length]
```

Comment:
```suggestion
        sin = cache_kwargs.get("sin")[-self.window_length:]
```
would that not make more sense? since that's the side we split the cache from no?

Reply:
I'll run some tests for this

Reply:
Updated my previous message with my findings

Reply:
@ArthurZucker @tomaarsen that's the beauty of sin and cos: the rerotation is based on the relative angles, and the relative angles are the same regardless of the side we slice 🙌 

If you place a debugger, you can see that the sliced tensors are different, but the rerotation coefficients are exactly the same!